server/sonar-server/src/test/java/org/sonar/server/platform/monitoring/CeStateMonitorTest.java

```diff
@@ -45,9 +46,16 @@ public void testName() {
   public void testAttributes() {
     when(jmxConnectionFactory.create(ProcessId.COMPUTE_ENGINE).getSystemState()).thenReturn(ImmutableSortedMap.<String, Object>of(
       "foo", "foo_val", "bar", "bar_val"));
-    LinkedHashMap<String, Object> attributes = underTest.attributes();
-    assertThat(attributes).containsExactly(
+    Optional<Map<String, Object>> attributes = underTest.attributes();
+    assertThat(attributes.get()).containsExactly(
       MapEntry.entry("bar", "bar_val"),
       MapEntry.entry("foo", "foo_val"));
   }
+
+  @Test
+  public void absent_attributes_if_CE_is_down() {
+    when(jmxConnectionFactory.create(ProcessId.COMPUTE_ENGINE)).thenReturn(null);
+    Optional<Map<String, Object>> attributes = underTest.attributes();
+    assertThat(attributes.isPresent()).isFalse();
```

Comment:
tip: you can statically import `org.assertj.guava.api.Assertions.assertThat` and then write `assertThat(attributes).isPresent()`